numpy/typing/tests/data/fail/modules.py

```diff
@@ -1,3 +1,318 @@
 import numpy as np
 
 np.testing.bob  # E: Module has no attribute
+np.DataSource  # E: Module has no attribute
```

Comment:
Ah I feel I was not quite clear enough here: 

The tests in `fail/` directory are for expressions which are supposed to fail, the relevant mypy error message being appended as a comment (_e.g._ `np.testing.bob  # E: Module has no attribute`). With your additions in https://github.com/numpy/numpy/pull/17406/commits/0996962236506342d16730580e126efe2dfbf98d you're effectivelly saying that, for example, the `np.DataSource` object does not exist and that trying the get it will/should result in an error.

Can you revert https://github.com/numpy/numpy/pull/17406/commits/0996962236506342d16730580e126efe2dfbf98d and replace it with something along the likes of:
``` python
np.bob  # E: Module has no attribute
```

Reply:
Thanks for the explanation.
Ah should I describe non-existent object Regardless of what is added to \_\_init\_\_.pyi ?

I want to be clear why do that when  `np.testing.bob  # E: Module has no attribute` already exists in fail/moodules.py ?
I think I've looked over how to contribute numpy contribution , but I'm sorry if I've missed something.

Reply:
Ah, I understood. 
It means that it is necessary to add test for top-level namespace.

Reply:
Done!